### PR TITLE
Update transaction properties

### PR DESCRIPTION
**Pull request by rtrochet**
## What

Updates the properties of a transaction to match the current state at https://plaid.com/docs/api/products/transactions/#transactionssync.

## How

Adds / removes properties from the Plaid YAML file. Updates test files (I think??) to match.

## User Impact

None yet; I didn't add `transaction` as a syncable stream since pagination and incremental sync are still broken.

## Can this PR be safely reverted and rolled back?

- [x] YES 💚
- [ ] NO ❌

### CODE REVIEW

File: airbyte-integrations/connectors/source-plaid/source_plaid/manifest.yaml [hunk 1]
```diff
@@ -240,23 +265,28 @@ streams:
...
+              primary:
+                type:
+                  - string
+              detailed:
+                type:
+                  - string
+              confidence_level:
+                type:
+                  - string
+                  - "null"
```
Comment (pilot-jasonslchen) [inline airbyte-integrations/connectors/source-plaid/source_plaid/manifest.yaml:283]: nit: do we want to add 
```
type:
- object
- "null"
```
 at the `properties level`? One indent in from `personal_finance_category`

File: airbyte-integrations/connectors/source-plaid/source_plaid/manifest.yaml [hunk 2]
```diff
@@ -145,14 +145,38 @@ streams:
             type:
               - string
               - "null"
+          counterparties:
```
Comment (pilot-jasonslchen) [inline airbyte-integrations/connectors/source-plaid/source_plaid/manifest.yaml:148]: Leaving a comment here, it looks like `category` and `category_id` are deprecated in `transactions/sync` do we also want to remove those?

**Review by pilot-jasonslchen — COMMENTED** [review]
One question before I approve!

File: airbyte-integrations/connectors/source-plaid/source_plaid/manifest.yaml (hunk 1 above)
Comment (rtrochet) [reply airbyte-integrations/connectors/source-plaid/source_plaid/manifest.yaml:283]: We sure do! Thanks for spotting that.

File: airbyte-integrations/connectors/source-plaid/source_plaid/manifest.yaml (hunk 2 above)
Comment (rtrochet) [reply airbyte-integrations/connectors/source-plaid/source_plaid/manifest.yaml:148]: I went back and forth about this—our code currently reads them, so I settled on continuing to import them for now. But I'll file a ticket to remove them once we fix our bank sync code.

**Review by pilot-jasonslchen — APPROVED** [review]
Looks good! 🚢

**Pull request merged**